@here/harp-mapview/lib/text/ScreenSpaceRenderer.ts

```diff
@@ -8,8 +8,8 @@ import { TextCanvas } from "@here/harp-text-canvas";
 
 import { PoiRenderer } from "../poi/PoiRenderer";
 
-export interface TextCanvasRenderer {
-    fontCatalog: string;
-    textCanvas: TextCanvas;
-    poiRenderer: PoiRenderer;
```

Comment:
TextElementRenderer always gets a PoiRenderer and always sets it for all TextCanvasRenderer instances. You can make this member required. That way you don't have to test for undefined in many places in TextElementsRenderer.

Actually even simpler. If we now only have one PoiRenderer (thank you for that!), there's no need to store here a reference, TextElementsRenderer can always get this.m_poiRenderer. And then probably you don't need to rename this interface cause it only handles text related stuff.

Reply:
I actually just this minute pushed a change exactly doing this :)